js/src/logger.ts

```diff
@@ -927,6 +995,29 @@ with a Blob/ArrayBuffer, or run the program on Node.js.`,
       return new LazyValue(async () => new Blob([data]));
     }
   }
+
+  private async getMetadata() {
```

Comment:
If we are re-using this between the downloader and the status-getter, should we make it a lazy value so that it's cached? Otherwise maybe it would be slightly better to call `/attachment/status` in the status-getter?

Reply:
Not caching with LazyValue is intentional here because the status could change during the lifetime of the Attachment object. For example if a dataset is being iterated concurrently with some attachment uploads. In that case, I am imagining a caller may want to retry reading the attachment later. 

I agree with calling `/attachment/status` if the `downloadUrl` is not needed, so that the backend request handler can potentially return faster.

Reply:
Hmm interesting. I see how you are throwing an exception inside `initDownloader` to cancel the LazyValue caching in case the file is not finished uploading, so it does make sense to fetch the status independently.